src/bun.js/child_process.exports.js

```diff
@@ -890,6 +894,7 @@ export class ChildProcess extends EventEmitter {
       err.spawnargs = ArrayPrototypeSlice.call(this.spawnargs, 1);
       this.emit("error", err);
     } else {
+      await this.#handleExited;
```

Comment:
`exited` will return a resolved Promise when already exited

This is a good usecase for `Bun.peek`.

```suggestion
    const maybeExited = Bun.peek(this.#handleExited);
    if (maybeExited === this.#handleExited) {
       this.exitCode = await this.#handleExited;
    } else {
       this.exitCode = maybeExited;
    }
       
```

Reply:
Sick, I gotta get into the habit of thinking about Bun.peek use cases 🔥